corehq/apps/domain/decorators.py

```diff
@@ -287,8 +293,8 @@ def _inner(request, domain, *args, **kwargs):
     return _outer
 
 
-def _two_factor_required(domain, couch_user):
-    return domain.two_factor_auth and couch_user.is_web_user() and not couch_user.two_factor_disabled
+def _two_factor_required(allow_two_factor, domain, couch_user):
+    return allow_two_factor and domain.two_factor_auth and not couch_user.two_factor_disabled
```

Comment:
nit: I was a little confused about what `allow_two_factor` means - I think it's "this view requires two factor if enabled on the domain", but that's hard to fit into a variable name.  Another option would be (`not`) `skip_two_factor`.  Take your pick.